lib/tasks/lumen.rake

```diff
@@ -737,6 +738,34 @@ where works.id in (
 
     puts "#{date_time_task.call} Finishing the task"
   end
+
+  desc 'Precomputation for word cloud'
+  task sampling_and_weight_calculation: :environment do
+    sample_count = 100
+    buckets = 5
+    total_notices = Notice.get_approximate_count
+    word_frequency = Hash.new
+    filter = Stopwords::Snowball::Filter.new "en"
+    Notice::TYPES.each do |type|
+      topic_count = type.constantize.count
+      offset_val = topic_count / buckets
+      sample_size = ((topic_count * 1.0 / total_notices) * sample_count).ceil
+      for i in 0..buckets - 1 do
```

Comment:
Nesting loops/conditionals this deep is going to make this code hard to read and modify over time. (Google for Sandi Metz's "squint test".)

Some things that would help readability here:

* make `notice_type_subsample` a function -- that lets you hide the messy bit of math, and makes it easier for us to change its implementation in future if we decide other math is better
* Instead of `if text.present?` with a nested `if` clause, use a guard clause -- `next unless text.present?` -- then everything following is at the same level of indentation
* instead of `word_frequency = Hash.new`, initialize it with a default -- `word_frequency = Hash.new(0)`. This means you won't have to do the `has_key?` check -- `word_frequency[:my_key]` will equal `0` by default if you haven't yet assigned it -- so you can just do `word_frequency[:my_key] += 1` without checking for the existence of the key.